src/zabapgit_forms.prog.abap

```diff
@@ -119,6 +100,33 @@ FORM branch_popup TABLES   tt_fields TYPE ty_sval_tt
 
 ENDFORM.                    "branch_popup
 
+FORM package_popup TABLES   tt_fields TYPE ty_sval_tt
+                   USING    pv_code TYPE clike
+                   CHANGING cs_error TYPE svale
+                            cv_show_popup TYPE c
+                   RAISING  lcx_exception ##called ##needed.
+* called dynamically from function module POPUP_GET_VALUES_USER_BUTTONS
+
+  DATA: ls_package_data TYPE scompkdtln.
+
+  FIELD-SYMBOLS: <ls_fbranch> LIKE LINE OF tt_fields.
+
+  CLEAR cs_error.
+
+  IF pv_code = 'COD1'.
+    cv_show_popup = abap_true.
+
+    ls_package_data = lcl_popups=>popup_to_create_package( ).
+
+    lcl_sap_package=>create( ls_package_data ).
```

Comment:
if the user presses cancel in the popup, then `ls_package_data` is initial and no package should be created